test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -87,10 +88,14 @@ private void assertCommandBehavior(String inputCommand,
             assertEquals(lastShownList, r.getRelevantPersons().get());
         }
 
+        Command cmd = new Parser().parseCommand(inputCommand);
+
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (cmd.isMutating()) {
```

Comment:
Good job in locating the right place to update the unit test. How about the else branch? What should you check is the method is not mutating?

Reply:
Updated the else branch